SupportYourLocals.Data/CSVData.cs

```diff
@@ -5,73 +5,97 @@
+        {
+            return false;
+        }
+        public void SaveRegisterData(String password, String username)
         {
-            // Checking person's ID
-            List<double> personID = new List<double>();
-            using (TextFieldParser csvParser = new TextFieldParser(filePath))
+            StringBuilder csv = new StringBuilder();
+
+            if (!File.Exists(filePath))
             {
-                csvParser.CommentTokens = new[] { "#" };
-                csvParser.SetDelimiters(new[] { "," });
-                csvParser.HasFieldsEnclosedInQuotes = true;
+                csv.AppendLine("Username, Password, ID");
+            }
 
-                // Skip the row with the column names
-                csvParser.ReadLine();
+            String salt = CreateSalt(saltSize);
 
-                while (!csvParser.EndOfData)
-                {
-                    // Read current line fields, pointer moves to the next line.
-                    string[] fields = csvParser.ReadFields();
-                    personID.Add(double.Parse(fields[3]));
-                }
-            }
+            //String hashedPassword = BitConverter.ToString(GenerateHash(password, salt)).Replace("-", "");
+
+            var newLine = "{0},{1},{2},{3}".Format(username, GenerateHash(password, salt), salt, personsID);
+
+            csv.AppendLine(newLine);
 
-            // Setting person's ID
-            personsID += personID.Count;
-            return personsID;
+            File.AppendAllText(filePath, csv.ToString());
         }
+        public bool CompareHash(string attemptedPassword, byte[] hash, string salt)
+        {
+            string base64Hash = Convert.ToBase64String(hash);
+            string base64AttemptedHash = Convert.ToBase64String(GenerateHash(attemptedPassword, salt));
 
-        public static void SetMarkers(List<double> listXCoord, List<double> listYCoord, List<int> listPersonsID)
+            return base64Hash == base64AttemptedHash;
+        }
+
+        public String CreateSalt(int size)
         {
-            using (TextFieldParser csvParser = new TextFieldParser(filePath))
-            {
-                csvParser.CommentTokens = new[] { "#" };
-                csvParser.SetDelimiters(new[] { "," });
-                csvParser.HasFieldsEnclosedInQuotes = true;
+            var rng = new System.Security.Cryptography.RNGCryptoServiceProvider();
+            var buff = new byte[size];
+            rng.GetBytes(buff);
+            return Convert.ToBase64String(buff);
+        }
 
-                // Skip the row with the column names
-                csvParser.ReadLine();
+        public byte[] GenerateHash(String password, String salt)
+        {
+            byte[] bytes = System.Text.Encoding.UTF8.GetBytes(password + salt);
+            System.Security.Cryptography.SHA256Managed sha256hashstring = new System.Security.Cryptography.SHA256Managed();
```

Comment:
Wherever the class type is obvious, for example when using the new operator (like here), it's better to just use var instead of specifying the exact type.